src/viam/utils.py

```diff
@@ -184,3 +188,12 @@ async def wait(self) -> None:
     def count(self) -> int:
         with self._lock:
             return self._count
+
+
+async def to_thread(func, /, *args, **kwargs):
+    if sys.version_info >= (3, 9):
+        return await asyncio.to_thread(func, *args, **kwargs)
+    loop = asyncio.events.get_running_loop()
+    ctx = contextvars.copy_context()
+    func_call = functools.partial(ctx.run, func, *args, **kwargs)
+    return await loop.run_in_executor(None, func_call)
```

Comment:
I might be helpful to add a comment in code with a link to this logic as you did the PR